modules/tinymce/src/plugins/table/test/ts/browser/NestedFakeSelectionTest.ts

```diff
@@ -0,0 +1,436 @@
+import { Cursors, Keys, Mouse } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyContentActions, TinyDom, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { Attribute, SelectorFilter, SugarElement } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+interface Scenario {
+  readonly content: string;
+  readonly selection: Cursors.CursorPath;
+  readonly expectedSelectedCells: string[];
+  readonly keyDirection: number;
+}
+
+describe('browser.tinymce.plugins.table.NestedFakeSelectionTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'table',
+    indent: false,
+    valid_styles: {
+      '*': 'width,height,vertical-align,text-align,float,border-color,background-color,border,padding,border-spacing,border-collapse'
+    },
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ]);
+
+  // The critical part is the target element as this is what Darwin (MouseSelection.ts) uses to determine the fake selection
+  const selectWithMouse = (start: SugarElement<Element>, end: SugarElement<Element>) => {
```

Comment:
Should some of these be moved over to `TableTestUtils`? as atm they are duplicated with the `FakeSelectionTest`.

I'm mainly thinking the following:
- `selectWithMouse`
- `selectWithKeyboard`

and then maybe these, though they aren't duplicated?
- `getSelectedCells`
- `assertSelectedCells`

Reply:
Good idea. This has been done now.